modules/frontend/querysharding.go

```diff
@@ -174,9 +196,31 @@ func mergeResponses(rrs []RequestResponse) (*http.Response, error) {
 	}
 
 	if errCode == http.StatusOK {
+		if marshallingFormat == util.ProtobufTypeHeaderValue {
```

Comment:
if there's a sane way to combine this with the querier code that also uses the Accept header to choose how to marshal, please do.  otherwise this is fine.

Reply:
Maybe could be cleaned up like:

```
var r Reader

if format == Proto {
 r := protoThing
} else {
  r := jsonThing
}

return &http.Response{  
  Body: Use r
 }
```

Reply:
Have been thinking about a function like so - 

```
func MarshalTraceObject(trace *tempopb.Trace, marshallingFormat string) ([]byte, error)
```

But while this could be useful at the querier, at the frontend we receive the response in bytes, its not beneficial to unmarshal that into the object and then remarshal it at the function if we're going to return proto bytes anyway..